lib/x509.mli

```diff
@@ -477,12 +475,12 @@ module Certificate : sig
 
   (** [decode_pkcs1_digest_info buffer] is [hash, signature], the hash and raw
       signature of the given [buffer] in ASN.1 DER encoding, or an error. *)
-  val decode_pkcs1_digest_info : Cstruct.t ->
-    (Mirage_crypto.Hash.hash * Cstruct.t, [> `Msg of string ]) result
+  val decode_pkcs1_digest_info : string ->
+    ([ `MD5 | `SHA1 | `SHA224 | `SHA256 | `SHA384 | `SHA512 ] * string, [> `Msg of string ]) result
```

Comment:
looking at this, I'm unsure. sometimes Digestif.hash' is used, but here and below it is explicit. Is this intentional (sorry I lost context and am on a mobile phone screen)

Reply:
`Digestif.hash'` has more variants that this one. On the interface level, if we use `Digestif.hash'`, this would mean we'd have to handle cases like `RIPEMD160`, for example, in the implementation. To avoid unnecessary cases, this is why we make variants explicit.